dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -1021,9 +1022,9 @@ public Knowledge visitSyncBarrier(Relation sync_bar) {
         @Override
         public Knowledge visitSyncFence(Relation sync_fen) {
             Set<Tuple> may = new HashSet<>();
-            List<Fence> fenceEvents = program.getThreadEvents(Fence.class);
-            for (Fence e1 : fenceEvents) {
-                for (Fence e2 : fenceEvents) {
+            List<GenericVisibleEvent> fenceEvents = program.getThreadEvents(GenericVisibleEvent.class).stream().filter(e -> e.hasTag(Tag.FENCE)).toList();
+            for (GenericVisibleEvent e1 : fenceEvents) {
+                for (GenericVisibleEvent e2 : fenceEvents) {
                     if (e1.hasTag(Tag.PTX.SC) && e2.hasTag(Tag.PTX.SC) && !exec.areMutuallyExclusive(e1, e2)) {
```

Comment:
Same as above. But you can additionally filter for `Tag.PTX.SC`, because the loop only ever cares about those events.